examples/tutorials/CartPole/CartPole_Tutorial.md

```diff
@@ -0,0 +1,516 @@
+```
+
+The second part of is to override the step and reset methods from the `AbstractEnv` class. For the step method, no additions are required. For the reset method, we need to restart the scene using `start_scene` and return the first observation. It might also be needed to update some parameters or configs like the randomized goal position in the case of some environments.
+
+```python
+    def step(self, action):
+        return super().step(action)
+
+    def reset(self):
+        """Reset simulation.
+
+        Note:
+        ----
+            We launch a client to create the scene. The scene of the program is
+            client_<scene>Env.py.
+
+        """
+        super().reset()
+
+        self.config.update({'goalPos': self.goal})
+
+        obs = start_scene(self.config, self.nb_actions)
+        
+        return np.array(obs['observation'])
+```
+
+
+# Step 2: Create the Environment's Toolbox
+
+The next step is to implement the methods and functions necessary for the RL algorithms to work. Three essential parts need to be implemented for this, defining and applying the actions, defining and calculating the reward, and getting the new observation from the simulation and updating it. One additional component that is relevant to some environments is a `GoalSetter` to define and update a target goal if necessary.
+
+First, we need to creat a new file `CartPoleToolbox`. For each of the parts that need to be implemented, some components are required to be defined to make the SOFA scene compatible with SofaGym.
+- Actions:
+  - startCmd function
+- Reward:
+  - getReward function
+  - rewardShaper "Reward" class
+- Observation:
+  - getState function
+  - getPos function
+  - setPos function
+- Goal:
+  - goalSetter "GoalSetter" class
+
+
+## Actions
+The possible actions that could be applied by the agent must be defined. The CartPole environment has two actions as discussed in the previous step, move the cart left or right. This could be achieved by applying a constant force field on the cart with a positive value (right direction) or a negative value (left direction) depednding on the chosen action 0 or 1, respectively.
+
+We can define an `applyAction` class that inherits from `Sofa.Core.Controller`, which allows us to add this component to the scene and make it update the value of the `constantForceField` component during the simulations steps. We initialize an `incr` variable to define the value of the force change between two consecutive steps.
+
+The `compute_action` method sets the actual value of the action to take based on the value returned by the RL algorithm. The `_move` method is used to update the force field applied on the cart by changing the x value of the `constantForceField` object of the cart to control the motion of the cart left and right. Finally, the `apply_action` method will be used to apply the action by the agent.
+
+```python
+class applyAction(Sofa.Core.Controller):
```

Comment:
According to PEP8, class names must start with a capital letter

Reply:
I edited this in the tutorial file but not in the code itself, but since there are more instances of this over the whole code, I think it would be better for me to clean the code and fix these naming conventions in a separate PR.

Reply:
Ok no problems